src/main/java/gov/usgs/wma/waterdata/groundwater/S3Bucket.java

```diff
@@ -38,10 +37,13 @@ public String getKeyName() {
 	@Override
 	public void close() throws Exception {
 		try {
-			writer.close();
-			sendS3();
+
+			if (writer != null) {
+				writer.close();
+				sendS3();
```

Comment:
I am weirded out by this - Why do we want sending the file to be a side effect of closing the file?  This is out of the hands of the caller - maybe the there is a better way?  It seems like the send method should be an action that the caller gets to determine they want to call.

Reply:
There is a "reason" and it might not be a good one "any longer." It's complicated but I will attempt in text.
I started writing this thinking that "for sure there will be a trivial means to stream a file to S3" and wrote it as if I was going to be streaming it. You cannot stream to S3 unless you perform some nasty gyrations that I think required threading. Well, at this point the pattern is as tho the S3 module is streaming, but it has to send it to a file and push the file. So in my "cleverness" it is acting like a stream. The S3 is open, data is sent, and then it is closed. Hidden in the impl (where I think it belongs) is that it secretly streams to a temp file and then sends the temp file to S3 as AWS requires. It is not all that weird, it is purposefully hiding from the outer impl that we have to have a temp file.